components/builder-api/src/server/resources/channels.rs

```diff
@@ -553,6 +559,38 @@ fn get_packages_for_origin_channel_package_version(req: HttpRequest,
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+fn get_packages_for_origin_channel_package_version_head(req: HttpRequest,
+                                                        path: Path<(String,
+                                                              String,
+                                                              String,
+                                                              String)>,
+                                                        pagination: Query<Pagination>)
+                                                        -> HttpResponse {
+    let (origin, channel, pkg, version) = path.into_inner();
+    let channel = ChannelIdent::from(channel);
+
+    let ident = PackageIdent::new(origin, pkg, Some(version), None);
+
+    match do_get_channel_packages(&req, &pagination, &ident, &channel) {
+        Ok((_packages, count)) => {
```

Comment:
This is a take it or leave it;  It reads a bit cleaner (to me), but isn't obvious what the `0` is. We could do this in the match instead. Ex: 
```
Ok((_, 0)) => { HttpResponse::new(StatusCode::NOT_FOUND) },
Ok((_packages, _count)) => { 
  HttpResponse::Ok().header(http::header::CONTENT_TYPE, headers::APPLICATION_JSON)
                                  .header(http::header::CACHE_CONTROL,
                                          headers::Cache::NoCache.to_string())
                                  .finish()
},
....
```

Reply:
Oh I like that! Will get this in today.